service/src/main/kotlin/app/cash/backfila/client/EnvoyClientServiceClientProvider.kt

```diff
@@ -2,23 +2,25 @@ package app.cash.backfila.client
 
 import app.cash.backfila.client.Connectors.ENVOY
 import com.squareup.moshi.Moshi
+import misk.client.EnvoyClientEndpointProvider
 import javax.inject.Inject
 import javax.inject.Named
 import javax.inject.Singleton
 import misk.client.HttpClientConfigUrlProvider
 import misk.client.HttpClientEndpointConfig
 import misk.client.HttpClientEnvoyConfig
 import misk.client.HttpClientFactory
+import misk.client.HttpClientsConfig
 import misk.moshi.adapter
 import retrofit2.Retrofit
 import retrofit2.adapter.guava.GuavaCallAdapterFactory
 import retrofit2.converter.wire.WireConverterFactory
 
 @Singleton
 class EnvoyClientServiceClientProvider @Inject constructor(
-  @Named(ENVOY) private val defaultHttpClientEndpointConfig: HttpClientEndpointConfig,
+  private val httpClientsConfig: HttpClientsConfig,
   private val httpClientFactory: HttpClientFactory,
-  private val httpClientConfigUrlProvider: HttpClientConfigUrlProvider,
+  private val envoyClientEndpointProvider: EnvoyClientEndpointProvider,
```

Comment:
@shellderp what do you think of replacing connectorExtraData with just a url field?

Reply:
sounds trivial for http

```
data class HttpConnectorData(val url: String)

data class EnvoyConnectorData(val clusterType: String)
```

for envoy would `clusterType` be the "url"?

Reply:
seems reasonable, ultimately it's just a callback url

Reply:
Well I was thinking of just replacing the data type completely. Although now I'm thinking that is just something to do when we run into the next client type?

Unfortunately the retrofit clients are still a bit different between the envoy and http clients. Unless you think that should be specified too.

Reply:
@mpawliszyn Could you clarify why retrofit clients are different? From what I've seen, to build a retrofit client you still go to HttpClientsConfig to ask it for config, and they you build the client from that. Therefore, same approach still applies - as long as you know the URL you can build the client. Or am I missing something?

Reply:
yea I meant replace data structure with url. exactly what does the url look like for the envoy client though? I suppose that's not my problem :p